spec/support/shared_examples/phone/rate_limitting.rb

```diff
@@ -42,11 +45,32 @@
 
   it 'limits the number of times the user can enter an OTP' do
     visit_otp_confirmation(delivery_method)
-    (max_confirmation_attempts - 1).times do
+    (max_confirmation_attempts - 1).times do |number_of_times|
       fill_in :code, with: '123456'
       click_submit_default
-
-      expect(page).to have_content(t('two_factor_authentication.invalid_otp'))
+      count_remaining = max_confirmation_attempts - user.reload.second_factor_attempts_count
+      expect(page).to have_content(
+        t('two_factor_authentication.invalid_otp'),
+      )
+      if count_remaining >= min_attempts
+        expect(page).to have_no_content(
+          strip_tags(
+            t(
+              'two_factor_authentication.attempt_remaining_warning_html',
+              count: count_remaining,
+            ).to_s,
```

Comment:
Is the `to_s` necessary here? (I would expect the same content being asserted in the `if` and `else` blocks)

```suggestion
            ),
```